src/Templates/ODataT4CodeGenerator.cs

```diff
@@ -1855,7 +1874,7 @@ internal void WriteNamespace(string fullNamespace)
         {
             if(context.GenerateMultipleFiles)
             {
-                context.MultipleFilesManager.StartNewFile($"ExtensionMethods{(this.context.TargetLanguage == LanguageOption.VB ? ".vb" : ".cs")}",false);
+                context.MultipleFilesManager.StartNewFile($"{(this.context.NamespacesInModel.Length > 1 ? $"{fullNamespace}." : string.Empty)}ExtensionMethods{(this.context.TargetLanguage == LanguageOption.VB ? ".vb" : ".cs")}", false);
```

Comment:
```suggestion
                var ns = this.context.NamespacesInModel.Length > 1 ? $"{fullNamespace}." : string.Empty;
                var fileExtension = this.context.TargetLanguage == LanguageOption.VB ? ".vb" : ".cs";
                context.MultipleFilesManager.StartNewFile($"{ns}ExtensionMethods{fileExtension}", false);
```
@unchase For readability